src/test/scala/chiselTests/experimental/hierarchy/InstanceSpec.scala

```diff
@@ -298,6 +298,21 @@ class InstanceSpec extends ChiselFunSpec with Utils {
       annos should contain(MarkAnnotation("~Top|Top/i:HasEither>x".rt, "xright"))
       annos should contain(MarkAnnotation("~Top|Top/i:HasEither>y".rt, "yleft"))
     }
+    it("3.12: should properly support val modifiers") {
+      class SupClass extends Module {
+        val value = 10
+        val overriddenVal = 10
+      }
+      @instantiable class SubClass() extends SupClass {
+        // This errors
```

Comment:
you could add a `should not compile` check if you want to check the "this errors" case

Reply:
what about `final`?

Reply:
Yeah I tried it, but I think that functionality interacts incorrectly with the `@public` macros because it itself is a macro, so it ends up not working.